projects/grpc-client-server/prober_server/main.go

```diff
@@ -22,26 +22,64 @@ type server struct {
 	pb.UnimplementedProberServer
 }
 
-func (s *server) DoProbes(ctx context.Context, in *pb.ProbeRequest) (*pb.ProbeReply, error) {
-	// TODO: support a number of repetitions and return average latency
-	start := time.Now()
-	_, _ = http.Get(in.GetEndpoint())	// TODO: add error handling here and check the response code
-	elapsed := time.Since(start)
-	elapsedMsecs := float32(elapsed / time.Millisecond)
-
-	return &pb.ProbeReply{LatencyMsecs: elapsedMsecs}, nil
+func (s *server) DoProbes(ctx context.Context, request *pb.ProbeRequest) (*pb.ProbeReply, error) {
+	// get the number of probes from the probe request
+	numberOfProbes := request.GetNumberOfProbes()
+	// fmt.Printf("DoProbes | numberOfProbes %v\n", numberOfProbes)
+
+	// initialise a total time
+	var totalTime time.Duration
+
+	// support a number of repetitions and return average latency
+	for i := 0; i < int(numberOfProbes); i++ {
+		startTime := time.Now()
+		// fmt.Printf("DoProbes | startTime %v\n", startTime)
+
+		// make the request to the endpoint
+		response, err := http.Get(request.GetEndpoint())
+
+		// if the request errors
+		if err != nil {
+			return nil, fmt.Errorf("error: DoProbes request failed: %v", err)
+		}
+
+		// remember to close the response body
+		defer response.Body.Close()
+
+		// if the request status code is not OK
+		if response.StatusCode != http.StatusOK {
+			return nil, fmt.Errorf("error: response was not OK: %v", err)
```

Comment:
Same question as above about errors

Reply:
how would i both "error" correctly
but at the same time store this as a "fail" (in some sort of count and maybe even store the err message itself) 

because i am not allowed to ignore the error?
or am i allowed to ignore the error here?

i could store the errors (including count) somehow and then do something with them user feedback wise

Reply:
Yeah, I think storing either a count of the errors, or the whole errors, and returning them to the user probably is useful!